File: Server/Persistance/Service/Service.cs
using Problem11.Model;
using Problem11.Repositories;
using System;
using System.Collections.Generic;

namespace Problem11.Service
{
    public class Service
    {
        public AngajatRepository angajati { get; set; }
        public CursaRepository curse { get; set; }
        public EchipaRepository echipe { get; set; }
        public InscriereRepository inscrieri { get; set; }
        public ParticipantRepository participanti { get; set; }
        public Service(AngajatRepository angajatRepository,CursaRepository cursaRepository,EchipaRepository echipaRepository,InscriereRepository inscriereRepository,ParticipantRepository participantRepository)
        {
            this.angajati = angajatRepository;
            this.curse = cursaRepository;
            this.echipe = echipaRepository;
            this.inscrieri = inscriereRepository;
            this.participanti = participantRepository;
        }
        public List<Angajat> getAllEmployees()
        {
            return this.angajati.findAll();
        }
        public void AddAngajat(Angajat angajat)
        {
            this.angajati.save(angajat);
        }
        public void AddCursa(Cursa cursa)
        {
            this.curse.save(cursa);
        }
        public void AddEchipa(Echipa echipa)
        {
            this.echipe.save(echipa);
        }
        public void AddInscriere(Inscriere inscriere)
        {
            this.inscrieri.save(inscriere);

        }
        public void AddParticipant(Participant participant)
        {
            this.participanti.save(participant);
        }
        public bool LocalLogin(String user,String pass)
        {
            return this.angajati.LocalLogin(user, pass);
        }
        public List<DTOBJCursa> GroupByCapacitate()
        {
            return this.curse.GroupByCapacitate();
        }
        public List<DTOBJPart> cautare(String numeEchipa)
        {
            return this.echipe.cautare(numeEchipa);
        }
        public void InscrirereParticipant(int capacitate,String numeParticipant,String numeEchipa)
        {
            int idCursa, idParticipant, idEchipa, idInscriere;
            idCursa = this.curse.findIdByCapacitate(capacitate);
            idEchipa = this.echipe.FindIdByName(numeEchipa);
            idParticipant = this.participanti.FindMaxId() + 1;
            idInscriere = this.inscrieri.FindMaxId() + 1;
            Participant nou = new Participant(idParticipant, numeParticipant, idEchipa);
            this.AddParticipant(nou);
            Inscriere noua = new Inscriere(idInscriere, idParticipant, idCursa);
            this.AddInscriere(noua);
        }
    }
}

File: Lab nertwork java/Lab1/Persistance/src/main/java/service/Service.java
package service;

import models.*;
import repos.repositories.*;

public class Service {
    private CursaRepository cursaRepository;
    private EchipaRepository echipaRepository;
    private OficiuRepository oficiuRepository;
    private ParticipantRepository participantRepository;
    private InscriereRepository inscriereRepository;

    public Service(CursaRepository cursaRepository, EchipaRepository echipaRepository, OficiuRepository oficiuRepository, ParticipantRepository participantRepository,InscriereRepository inscriereRepository) {
        this.cursaRepository = cursaRepository;
        this.echipaRepository = echipaRepository;
        this.oficiuRepository = oficiuRepository;
        this.participantRepository = participantRepository;
        this.inscriereRepository=inscriereRepository;
    }

    public Iterable<Oficiu> findAllEmployees(){
        return this.oficiuRepository.findAll();
    }
    public void addAngajat(Oficiu angajat){
        this.oficiuRepository.save(angajat);
    }
    public void addCursa(Cursa cursa){
        this.cursaRepository.save(cursa);
    }
    public void addEchipa(Echipa echipa){
        this.echipaRepository.save(echipa);
    }
    public void addParticipant(Participant participant){
        this.participantRepository.save(participant);
    }
    public void addInscriere(Inscriere inscriere){
        this.inscriereRepository.save(inscriere);
    }
    public void deleteAngajat(int id){
        this.oficiuRepository.delete(id);
    }
    public void deleteCursa(int id){
        this.cursaRepository.delete(id);
    }
    public void deleteEchipa(int id){
        this.echipaRepository.delete(id);
    }
    public void deleteParticipant(int id){
        this.participantRepository.delete(id);
    }
    public int angajatiSize(){
        return this.oficiuRepository.size();
    }
    public int curseSize(){
        return this.cursaRepository.size();
    }
    public int echipeSize(){
        return this.echipaRepository.size();
    }
    public int participantiSize(){
        return this.participantRepository.size();
    }
    public void updateAngajat(Oficiu angajatvechi, Oficiu angajatnou){
        this.oficiuRepository.update(angajatvechi.getId(),angajatnou);
    }
    public Iterable<DTOAngajat> findOthersEmployees(DTOAngajat angajat){
        return this.oficiuRepository.findOtherEmployees(angajat);
    }
    public void updateCursa(Cursa cursaveche, Cursa cursanoua){
        this.cursaRepository.update(cursaveche.getId(),cursanoua);
    }
    public void updateEchipa(Echipa echipaveche, Echipa echipanoua){
        this.echipaRepository.update(echipaveche.getId(),echipanoua);
    }
    public void updateParticipant(Participant participantvechi, Participant participantnou){
        this.participantRepository.update(participantvechi.getId(),participantnou);
    }

    public boolean login(String username,String parola){
        return oficiuRepository.localLogin(username,parola);
    }

    public Iterable<Cursa> findAllCursa(){
        return cursaRepository.findAll();
    }

    public Iterable<CursaDTO> findlAllCursaDto(){
        return cursaRepository.findCursaNrParticipanti();
    }

    public Iterable<ParticipantCursaDTO> findAllParticipantiCursaDto(String t){
        return participantRepository.findByTeam(t);
    }

    public Integer findIdCursa(Integer cap){
        return cursaRepository.findIdCursa(cap);
    }

    public void addInscriereParticipant(String nume,String numeEchipa,Integer capacitate){
        Integer idParticipant=participantRepository.getNewIdParticipant()+1;
        Integer idInscriere=inscriereRepository.getNewIdInscriere()+1;
        Integer idCursa=cursaRepository.findIdCursa(capacitate);
        Integer idEchipa=echipaRepository.findIdEchipa(numeEchipa);
        //Integer i=inscriereRepository.findIdInscriere(idParticipant,idCursa);
        Participant pnou = new Participant(idParticipant,nume,idEchipa);
        this.addParticipant(pnou);
        Inscriere inoua = new Inscriere(idInscriere,idParticipant,idCursa);
        this.addInscriere(inoua);
        }

}

